### DISCUSSION TITLE
Custom Action inside a Table Column

### DISCUSSION THREAD

**Discussion by RPotkowaVegasystems** [filamentphp/filament#12334] [Help]
### Package

Table builder

### Package Version

v3.2.47

### How can we help you?

I would like to create an action inside a table column. Like TextColumn and after an edit icon where i can trigger a custom action etc.

**Comment by leandrocfe** [comment]
TextColumn supports `->action()` and `->url()`. You can also create a row action: https://filamentphp.com/docs/3.x/tables/actions#row-actions

**Reply by RPotkowaVegasystems** [reply]
OK, it works. But it is possible only to make an Icon after that Value, that triggers the action? Not the value itself? Maybe placing the Action inside a suffis etc.?

**Comment by dave-mills** [comment]
I'm exploring a similar idea for a new project and came across your question. I think that if you want the text and icon in the same column, then there's no way to make *only* the icon trigger the action unless you write your own custom column type. 

An alternative is to have the icon as a separate IconColumn. You can set that column's label to an empty string and do some formatting to put the icon next to the text, so it sort-of looks like the same column. Here's a working example from my demo project: 

```php
TextColumn::make('name')
    ->sortable()
    ->searchable()
    ->alignEnd(), // align the text to the end of the column so it's next to the icon
IconColumn::make('name_edit')
    ->getStateUsing(fn() => true) // the column requires a state to be passed to it
    ->icon(fn(bool $state): string => 'heroicon-m-pencil-square') // always show the 'edit' icon
    ->label('')
    ->action(\Filament\Tables\Actions\Action::make('edit-name')
        ->form([
            TextInput::make('name'),
        ])
        ->fillForm(fn($record) => [
            'name' => $record->name,
        ])
        ->action(fn($record, $data) => $record->update($data))
    ),
```

https://github.com/filamentphp/filament/assets/5711101/c835c28b-fbcf-4005-87a9-b7c4ae619374
